src/server/editorServices.ts

```diff
@@ -4487,8 +4488,18 @@ export class ProjectService {
         }
 
         this.logger.info(`Enabling plugin ${pluginConfigEntry.name} from candidate paths: ${searchPaths.join(",")}`);
-        if (!pluginConfigEntry.name || parsePackageName(pluginConfigEntry.name).rest) {
-            this.logger.info(`Skipped loading plugin ${pluginConfigEntry.name || JSON.stringify(pluginConfigEntry)} because only package name is allowed plugin name`);
+        if (!pluginConfigEntry.name) {
+            this.logger.info(`Skipped loading plugin ${JSON.stringify(pluginConfigEntry)} because plugin name was not specified`);
+            return;
+        }
+
+        const parsedPackageName = parsePackageName(pluginConfigEntry.name);
+        if (
+            !parsedPackageName.packageName ||
+            normalizePath(parsedPackageName.rest) !== parsedPackageName.rest ||
+            ensurePathIsNonModuleName(parsedPackageName.rest) === parsedPackageName.rest
+        ) {
+            this.logger.info(`Skipped loading plugin ${pluginConfigEntry.name} because only package name is allowed plugin name`);
             return;
         }

```

Comment:
If we are allowing subpaths in package name, this is better test i think

```suggestion
        if (
            !pluginConfigEntry.name ||
            isExternalModuleNameRelative(pluginConfigEntry.name) ||
            /[\\/]\.\.?($|[\\/])/.test(pluginConfigEntry.name)
        ) {
            this.logger.info(`Skipped loading plugin ${pluginConfigEntry.name || JSON.stringify(pluginConfigEntry)} because only package name is allowed plugin name`);
            return;
        }
```

Reply:
@RyanCavanaugh @DanielRosenwasser do we need to ensure that this is "extensionless" as well?

Reply:
lgtm 🙌 I'll get this applied tomorrow. I didn't have a good overview of all the utilities available in the repo and wasn't sure whether a custom additional regex was acceptable, so apologies for the rough attempt 😅

re. extensions; personally, I'd disallow them to limit requires/imports to qualified `package.json:main`, `exports` entries, and `index` files only. Just to be on the safe side btw, but that might be a little controversial. Not sure